### PR TITLE
DOMA-1306 add account number validation, wrote tests

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Now, within one organization, you can create meters with the same accountNumber in different units, fix it

### CODE REVIEW

File: apps/condo/lang/en.json [hunk 1]
```diff
@@ -399,7 +400,8 @@
   "pages.condo.meter.MeterNumber": "Meter number",
   "pages.condo.meter.MeterPlace": "Meter place",
   "pages.condo.meter.NumberOfMeter": "Number of meter device",
-  "pages.condo.meter.MeterIsExistMessage": "Meter with same number exist in organization",
+  "pages.condo.meter.MeterWithSameNumberIsExistMessage": "Meter with same number exist in organization",
```
Comment (SavelevMatthew) [inline apps/condo/lang/en.json:403]: Missing translation?

**Review by SavelevMatthew — APPROVED** [review]

File: apps/condo/lang/en.json (hunk 1 above)
Comment (nomerdvadcatpyat) [reply apps/condo/lang/en.json:403]: MeterWithSameAccountNumberIsExistMessage was redundant

File: apps/condo/domains/meter/schema/Meter.js [hunk 2]
```diff
@@ -97,6 +97,38 @@ const Meter = new GQLListSchema('Meter', {
...
+                    const value = resolvedData[fieldPath]
+                    let metersWithSameAccountNumberInOrganization
+                    if (operation === 'create') {
+                        metersWithSameAccountNumberInOrganization = await MeterApi.getAll(context, {
+                            accountNumber: value,
+                            organization: { id: resolvedData.organization },
+                            deletedAt: null,
+                            OR: [
+                                { unitName_not: value },
+                                { property: { id_not: resolvedData.property } },
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/Meter.js:111]: is the resolvedData.property exists if we don't update the property field?

File: apps/condo/domains/meter/schema/Meter.js [hunk 3]
```diff
@@ -97,6 +97,38 @@ const Meter = new GQLListSchema('Meter', {
...
+                            organization: { id: existingItem.organization },
+                            deletedAt: null,
+                            OR: [
+                                { unitName_not: value },
+                                { property: { id_not: existingItem.property } },
+                            ],
+                        })
+                    }
+
+                    if (metersWithSameAccountNumberInOrganization && metersWithSameAccountNumberInOrganization.length > 0) {
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/Meter.js:127]: it's OK to have two meters with the same account number

File: apps/condo/domains/meter/schema/Meter.js (hunk 2 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/meter/schema/Meter.js:111]: it's create operation, property always exist

**Comment by pahaz** [comment]
It's better to add something like:
```
    kmigratorOptions: {
        constraints: [
            {
                type: 'models.UniqueConstraint',
                fields: ['organization', 'number'],
                condition: 'Q(deletedAt__isnull=True)',
                name: 'property_unique_organization_and_address',
            },
        ],
    },
```

File: apps/condo/domains/meter/schema/Meter.js (hunk 3 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/meter/schema/Meter.js:127]: yes, it's bad naming

File: apps/condo/domains/meter/schema/Meter.js [hunk 4]
```diff
@@ -97,6 +110,42 @@ const Meter = new GQLListSchema('Meter', {
...
+                            accountNumber: value,
+                            organization: { id: resolvedData.organization },
+                            deletedAt: null,
+                            OR: [
+                                { unitName_not: resolvedData.unitName },
+                                { property: { id_not: resolvedData.property } },
+                            ],
+                        })
+                    }
+                    else if (operation === 'update' && resolvedData.accountNumber !== existingItem.accountNumber) {
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/Meter.js:128]:
```suggestion
                    } else if (operation === 'update' && resolvedData.accountNumber !== existingItem.accountNumber) {
```

**Review by pahaz — APPROVED** [review]

**Pull request merged**